addons/web_editor/static/src/scss/wysiwyg_snippets.scss

```diff
@@ -2348,15 +2348,21 @@
                 }
             }
 
-            .o_overlay_background > * {
-                display: block !important;
-                top: 0 !important;
-                right: 0 !important;
-                bottom: 0 !important;
-                left: 0 !important;
-                transform: none !important;
-                max-width: unset !important;
-                max-height: unset !important;
+            .o_overlay_background {
+                // It may contain cloned elements with a z-index that affects the
+                // overlay buttons, we need to isolate the stacking context here.
+                isolation: isolate;
+
+                > * {
+                    display: block !important;
+                    top: 0 !important;
+                    right: 0 !important;
+                    bottom: 0 !important;
+                    left: 0 !important;
+                    transform: none !important;
+                    max-width: unset !important;
+                    max-height: unset !important;
```

Comment:
Seems OK for me to force the `z-index` to 0, since we [only use an empty clone](https://github.com/odoo/odoo/blob/16.0/addons/web_editor/static/src/js/editor/snippets.options.js#L7556-L7557) of the element (no child nodes with probably other `z-index` levels...) code updated :+1: